### PR TITLE
replace sonatype oss parent

### LINKED ISSUE #6: Move to Sonatype's updated recommendations for publishing artifacts to Maven Central
As a developer publishing artifacts to Sonatype's Maven Central
I want to use the recommended, non-deprecated way of doing this
So that my deployments are future proof for newer versions of Java and Maven

**Acceptance Criteria**

* https://github.com/HotelsDotCom/hotels-oss-plugin-config/ no longer uses org.sonatype.oss.oss-parent as a parent.
* The approach in http://central.sonatype.org/pages/apache-maven.html is used instead.
* At least one of our OSS projects using our parent is changed to use the above and tested with both SNAPSHOT and release builds.

### PR DESCRIPTION
**Pull request by courtsVII**
resolves #6

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -33,20 +28,63 @@
     </license>
   </licenses>
 
+  <properties>
+    <maven.javadoc.plugin.version>3.0.1</maven.javadoc.plugin.version>
+    <maven.source.plugin.version>3.0.1</maven.source.plugin.version>
+    <maven.gpg.plugin.version>1.6</maven.gpg.plugin.version>
+    <nexus.staging.maven.plugin.version>1.6</nexus.staging.maven.plugin.version>
```
Comment (massdosage) [inline pom.xml]: This should be 1.6.8 and where is the property used?

File: pom.xml [hunk 2]
```diff
@@ -33,20 +28,63 @@
     </license>
   </licenses>
 
+  <properties>
+    <maven.javadoc.plugin.version>3.0.1</maven.javadoc.plugin.version>
+    <maven.source.plugin.version>3.0.1</maven.source.plugin.version>
+    <maven.gpg.plugin.version>1.6</maven.gpg.plugin.version>
```
Comment (massdosage) [inline pom.xml:34]: Where is this used? i.e. you are missing the gpg plugin definition.

**Review by massdosage — CHANGES_REQUESTED** [review]
I'm not sure whether this will work without declaring the release plugin but I'd like to give it a go and see first, we can always add that in later if this fails when we try do a release build. We will need the other changes I mentioned in the comments though.

File: pom.xml [hunk 3]
```diff
@@ -33,20 +28,77 @@
...
-        <version>1.6.3</version>
+        <version>${nexus.staging.maven.plugin.version}</version>
         <extensions>true</extensions>
         <configuration>
           <serverId>sonatype-nexus-staging</serverId>
           <nexusUrl>https://oss.sonatype.org/</nexusUrl>
           <autoReleaseAfterClose>true</autoReleaseAfterClose>
         </configuration>
       </plugin>
+      <plugin>
```
Comment (ddcprg) [inline pom.xml:62]: before we would only execute this plugin during release, now they will execute during snapshot builds, do we want this behaviour?

**Comment by ddcprg** [comment]
there is a conflict :)

File: pom.xml (hunk 3 above)
Comment (massdosage) [reply pom.xml:62]: Uncertain, I think we need to try this out and decide whether we need to associate some things (like this) with a release profile. We'll probably need some trial and error with the new settings and this feels like a good project to try it on as it's so separate and small and we can then apply the lessons learned here to the hotels-oss-parent etc.

**Pull request merged**